src/lib/realmdb/RealmDB.js

```diff
@@ -407,6 +412,27 @@ class RealmDB implements DB, ProfileDB {
   async deleteProfile(): Promise<boolean> {
     return this.profiles.deleteOne({ user_id: this.user.id })
   }
+
+  /**
+   *
+   * @param {TransactionData} data
+   * @returns {Promise<any>}
+   */
+  // eslint-disable-next-line require-await
+  async addToOutbox(data: TransactionData): Promise<any> {
```

Comment:
as recipientPublicKey and txHash are key fields i suggest to pass them separately
e.g. async addToOutbox(recipientPublicKey: string, txHash: string, encrypted: string): Promise<any>